test/datatable.jl

```diff
@@ -342,6 +342,40 @@ module TestDataTable
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), [1]).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), 1).columns) == [1]
 
+    @testset "unstack nullable promotion" begin
+        long = DataTable(Any[repeat([:c, :d], inner=12),
+                             vcat(linspace(0, 1, 12), linspace(2, 1, 12)),
```

Comment:
> I took this example from the unstack docstring, replacing rand with linspace to avoid srand and random number issues.

Yes, but there's no reason to use such a complicated example here. Just a handful a values are enough to test that promotion works.

> The two tested types here are converting standard arrays to NullableArrays and CategoricalArrays to NullableCategoricalArrays,

But you should explicitly test for column types of the result. The current code only checks that it compares equal, which can happen with any nullable vector, not just `NullableCategoricalArray` (and in the future, maybe even with a non-nullable vector with the new `Nullable` implementation).